collect_app/src/main/java/org/odk/collect/android/utilities/SharedPreferencesUtils.java

```diff
@@ -43,7 +43,7 @@
 
     private final Context context = Collect.getInstance();
 
-    String getJSONFromPreferences(Collection<String> keys) throws JSONException {
+    public static String getJSONFromPreferences(Collection<String> keys) throws JSONException {
```

Comment:
Does this look fine to you? 

```
public static String getJSONFromPreferences(Collection<String> passwordKeys) throws JSONException {
        Collection<String> keys = new ArrayList<>(GENERAL_KEYS.keySet());
        keys.addAll(passwordKeys);
        JSONObject sharedPrefJson = getModifiedPrefs(keys);
        Timber.i(sharedPrefJson.toString());
        return sharedPrefJson.toString();
    }
```

Reply:
Much better. Thanks. That changes the order from before (adding password keys to general rather than the other way around), but perhaps the order doesn’t matter.